src/py/flwr/superexec/simulation.py

```diff
@@ -0,0 +1,140 @@
+from flwr.common.constant import RUN_ID_NUM_BYTES
+from flwr.common.logger import log
+from flwr.server.superlink.state.utils import generate_rand_int_from_bytes
+
+from .executor import Executor, RunTracker
+
+
+class SimulationEngine(Executor):
+    """Simulation engine executor.
+
+    Parameters
+    ----------
+    num_supernodes: Opitonal[str] (default: None)
+        Total number of nodes to involve in the simulation.
+    """
+
+    def __init__(
+        self,
+        num_supernodes: Optional[str] = None,
+    ) -> None:
+        self.num_supernodes = num_supernodes
+
+    @override
+    def set_config(
+        self,
+        config: Dict[str, str],
+    ) -> None:
+        """Set executor config arguments.
+
+        Parameters
+        ----------
+        config : Dict[str, str]
+            A dictionary for configuration values.
+            Supported configuration key/value pairs:
+            - "num-supernodes": str
+                Number of nodes to register for the simulation.
+        """
+        if not config:
+            return
+        if num_supernodes := config.get("num-supernodes"):
+            self.num_supernodes = num_supernodes
+
+        # Validate config
+        if self.num_supernodes is None:
+            log(
+                ERROR,
+                "To start a run with the simulation plugin, please specify "
+                "the number of SuperNodes. This can be done by using the "
+                "`--executor-config` argument when launching the SuperExec.",
+            )
+            raise ValueError("`num-supernodes` must not be `None`")
+
+    @override
+    def start_run(
+        self, fab_file: bytes, override_config: Dict[str, str]
+    ) -> Optional[RunTracker]:
+        """Start run using the Flower Simulation Engine."""
+        try:
+            if override_config:
+                raise ValueError(
+                    "Overriding the run config is not yet supported with the "
+                    "simulation executor.",
+                )
+
+            # Install FAB to flwr dir
+            _, fab_id = get_fab_metadata(fab_file)
+            fab_path = install_from_fab(fab_file, None, True)
+
+            # Install FAB Python package
+            subprocess.check_call(
+                [sys.executable, "-m", "pip", "install", "--no-deps", str(fab_path)],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+
+            # In Simulation there is no SuperLink, still we create a run_id
+            run_id = generate_rand_int_from_bytes(RUN_ID_NUM_BYTES)
+            log(INFO, "Created run %s", str(run_id))
+
+            fab_name = Path(fab_id).name
+
+            # Start Simulation
+            proc = subprocess.Popen(  # pylint: disable=consider-using-with
+                [
+                    "flower-simulation",
+                    "--client-app",
+                    f"{fab_name}.client:app",
```

Comment:
This needs to be read from the FAB itself (key: `flower.components.clientapp`). Users might rename `client` or `app`, or move the module in a sub-package.